server/src/limiter.rs

```diff
@@ -22,19 +22,15 @@ impl Default for Limiter {
 impl Limiter {
     pub fn should_limit(&self, plan: &Plan) -> bool {
         match plan {
-            Plan::Query(query) => {
-                let read_reject_list = self.read_reject_list.read().unwrap().clone();
```

Comment:
Clone is not useless here, it will release read_reject_list's lock.

Reply:
Yeah, I think you are right.

But in sense of whole function its useless as it's much faster to just iterate and check values instead of cloning them and then checking.

Reply:
Or maybe the better word is "inefficient" instead of "useless".

Reply:
I've updated the MR description.

Reply:
I feel OK with this refactor. Readers would not block each other and acquiring a write lock of `read_reject_list` is a fairly unfrequent operation. So I prefer to check values inside the read lock rather than clone them out.